docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,192 @@
+    if req.status_code != 200:
+        print('Error retrieving ADI: %s' % req.content)
+        return []
+
+    data = req.json()
+    return data['hits']
+
+
+def fetch_versions(conn, product_name):
+    """Fetch the version strings for a product from product_versions table"""
+
+    # FIXME(willkg): There's probably a better way to do this, but I can't think of it offhand.
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT version_string
+        FROM product_versions
+        WHERE
+            product_name=%s
+    """, (product_name,))
+    versions = set([record[0] for record in cursor if not record[0].endswith('b')])
+    return list(versions)
+
+
+def fetch_product_version_id(conn, product_name, version_string):
+    """Given a product name and a version_string, returns a product_version_id"""
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT product_version_id
+        FROM product_versions
+        WHERE
+            product_name=%s AND
+            version_string=%s
+    """, (product_name, version_string))
+    return cursor.fetchone()[0]
+
+
+def insert_adu(conn, adu_count, adu_date, product_name, os_name, version_string):
+    """Inserts ADU data into product_adu table"""
+    # Fetch the product_version_id we need
+    product_version_id = fetch_product_version_id(conn, product_name, version_string)
+
+    # Insert the data
+    cursor = conn.cursor()
+    try:
+        cursor.execute("""
+            INSERT INTO product_adu (adu_count, adu_date, os_name, product_version_id)
+            VALUES (%s, %s, %s, %s)
+        """, (adu_count, adu_date, os_name, product_version_id))
+    except psycopg2.IntegrityError as exc:
+        # Hitting an IntegrityError here probably means this data is in the table already, so we
+        # just skip it.
+        pass
+    conn.commit()
+
+
+def get_connection():
+    """Builds a connection using ConnectionContext environment variables
+
+    :returns: postgres connection
+
+    """
+    # This uses the same environment variables and defaults as postgres ConnectionContext
+    host = os.environ.get('resource.postgresql.database_hostname', 'localhost')
+    port = int(os.environ.get('resource.postgresql.database_port', '5432'))
+    username = os.environ.get('secrets.postgresql.database_username', 'breakpad_rw')
+    password = os.environ.get('secrets.postgresql.database_password', 'aPassword')
+    dbname = os.environ.get('resource.postgresql.database_name', 'breakpad')
+
+    local_config = {
+        'database_hostname': host,
+        'database_port': port,
+        'database_username': username,
+        'database_password': password,
+        'database_name': dbname
+    }
+    dsn = (
+        "host=%(database_hostname)s "
+        "dbname=%(database_name)s "
+        "port=%(database_port)s "
+        "user=%(database_username)s "
+        "password=%(database_password)s"
+    ) % local_config
+    return psycopg2.connect(dsn)
```

Comment:
One thing @milescrabill and I were talking about yesterday was how much redundancy we have in configuration regarding database configuration. I've got too many things on my plate to tackle that now before this PR lands, but it sounds like a good bug to write up to clean up Socorro.